### PR TITLE
Fix the compatibility of the quantization for MHA 

### PR DESCRIPTION
**Pull request by james77777778**
We should use `__call__` for sublayers in MHA to ensure compatibility with quantization.

This PR should fix the CI for KerasHub:
https://github.com/keras-team/keras-hub/actions/runs/12061895640/job/33634697425

cc @mattdangerw

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR. This was done to avoid mask-related warnings when calling the layer with a mask (basically, the dense layers don't support masking but are called with a tensor that has a mask associated with it -- it's fine because the mask is handled at the level of the MHA layer). https://github.com/keras-team/keras/commit/647554aad8518de07a30b00316f688f313b767f3

Can we suppress the warning some other way? Deleting the mask maybe?

**Comment by james77777778** [comment]
> Can we suppress the warning some other way? Deleting the mask maybe?

I made modifications to the MHA layer to handle the propagation and deletion of masks.
A new test has been added to ensure that no warnings are triggered.

**Review by fchollet — APPROVED** [review]
Wonderful -- thank you for the fix!

**Comment by mattdangerw** [comment]
Awesome! Thanks for finding this.

@james77777778 I think this means some quantization use cases are just broken for the 3.7 release right? When using an unaltered MHA layer + int8 quantization on the torch backend?

@fchollet might be worth doing a point release with this fix in it, I'll message you.

**Comment by james77777778** [comment]
> @james77777778 I think this means some quantization use cases are just broken for the 3.7 release right? When using an unaltered MHA layer + int8 quantization on the torch backend?
> 

It was not only broken on the torch backend but also on jax and tensorflow. They just didn’t cause the CI to fail.

More details:
We rely on https://github.com/keras-team/keras/blob/master/keras/src/ops/operation.py#L51-L54 to dispatch the `call` or `quantized_call`. If other layers directly invoke `Layer.call`, this dispatch mechanism breaks.
